algorithms/indexing/symmetry.py

```diff
@@ -136,126 +135,87 @@ def __init__(self, unit_cell=None, space_group=None, max_delta=5):
         self.target_symmetry_primitive = None
         self.target_symmetry_reference_setting = None
         self.cb_op_inp_ref = None
+        self.cb_op_inp_best = None
 
-        target_unit_cell = unit_cell
         target_space_group = space_group
         if target_space_group is not None:
             target_space_group = target_space_group.build_derived_patterson_group()
 
-        if target_unit_cell is not None or target_space_group is not None:
+        if unit_cell is not None:
 
-            if target_unit_cell is not None and target_space_group is not None:
-                self._setup_target_unit_cell_and_space_group(
-                    target_unit_cell, target_space_group
-                )
+            assert (
+                space_group is not None
+            ), "space_group must be provided in combination with unit_cell"
 
-            elif target_unit_cell is not None:
-                self.target_symmetry_reference_setting = crystal.symmetry(
-                    unit_cell=target_unit_cell, space_group=sgtbx.space_group()
+            if unit_cell is not None and target_space_group is not None:
```

Comment:
```suggestion
            if target_space_group:
```
we are in `unit_cell != None` if block